src/components/Textarea/index.tsx

```diff
@@ -0,0 +1,98 @@
+import type { CSSProperties, HTMLAttributes, ReactElement } from 'react'
+import styled from '@emotion/styled'
+import { useRef } from 'react'
+
+export interface TextAreaProps extends HTMLAttributes<HTMLTextAreaElement> {
+  label?: string
+  placeholder?: string
+  children?: string
+  guideMessage?: string
+  textAreaStyle?: CSSProperties
+  bgType?: 'filled' | 'ghost'
+  autoFocus?: boolean
+}
+
+interface StyledTextAreaProps {
+  isFilled: boolean
+}
+export const TextArea = ({
+  label = '',
+  placeholder = '내용을 입력하세요.',
+  children = '',
+  guideMessage = '',
+  bgType = 'filled',
+  textAreaStyle,
+  autoFocus,
+  ...props
+}: TextAreaProps): ReactElement => {
+  const textAreaRef = useRef<HTMLTextAreaElement | null>(null)
+  const textAreaDefaultHeight = '120px'
+  const isFilled = bgType === 'filled'
+
+  const handleResizeHeight = (): void => {
+    if (!textAreaRef || !textAreaRef.current) {
+      return
+    }
+    textAreaRef.current.style.height = textAreaDefaultHeight
+    textAreaRef.current.style.height = `${textAreaRef.current.scrollHeight}px`
+  }
```

Comment:
```suggestion
  autoFocus,
  onInput,
  ...props
}: TextAreaProps): ReactElement => {
  const textAreaRef = useRef<HTMLTextAreaElement | null>(null)
  const textAreaDefaultHeight = '120px'
  const isFilled = bgType === 'filled'

  const handleResizeHeight = (e: FormEvent<HTMLTextAreaElement>): void => {
    if (!textAreaRef || !textAreaRef.current) {
      return
    }
    textAreaRef.current.style.height = textAreaDefaultHeight
    textAreaRef.current.style.height = `${textAreaRef.current.scrollHeight}px`
    onInput?.(e)
  }
```
onInput이 넘어오는 경우가 있을 것 같아 고려해주면 좋을 것 같습니다!